doc/source/DBEP/DBEP07.rst

```diff
@@ -0,0 +1,60 @@
+===================
+ DBEP 7 : Document mutation
+===================
+
+
+.. contents::
+   :local:
+
+Status
+======
+
+**Discussion**
+
+
+Branches and Pull requests
+==========================
+
+
+Abstract
+========
+The motivation of this DBEP is the handling of document mutation.
+There is a need for post-acquisition modification of metadata.
+
+
+
+
+
+Proposed Implementations
+====================
+There are three ways that have been discussed for handling this.
+Which one should be used has not been decided on as of now.
+
+
+#. Copy the original data into a backup data store. Create a new start document
+   with a new uid with the updated metadata. Lutz didn't like this because this
+   isn't backwards compatible with his previous measurements.
+   #* Pros: Data is never mutated, less risk for data corruption
+   #* Cons: This is not very backwards compatible for beamline scientists. I.e.
+     retrieval of data by uid must be updated with the newest uid used.
+#. Modify the existing start document, and save the changes made in a separate
+   changes database.
+   #* Pros : The user is unaffected by the changes
+   #* Cons : Once the change is made, it is a little difficult to view the
+     documents without the change.
+#. Create a separate database (mongo collection, sqlite database, spreadsheet,
```

Comment:
We are going to want to get the corrected version far more often than we are going to want the original so it makes more sense to store the store the final state and compute the diff when we want it than to store the original + diffs and compute the current everytime we access it.

Interestingly, this is the ways that git (store full version, compute diffs) and svn (store diffs, compute full version) work.

Reply:
I agree with this. I will move the two points to "other" implementations and keep the second point (which is basically the implementation performed for resources)